[Bug]: Error desc not specifying what went wrong

### Actual Behavior


So i use:

    from openapi_core import validate_request

To validate request data versus schema:

    validate_request(request_data, self.spec)

Here's the snipped for request body:

     getRequest:
      type: object
      properties:
        type_of:
          type: string
        name:
          type: string

My request:
        r = requests.post(url='host/something/v012/get', data={'type_of': 'user', 'name': 'test'})
The error message raised is:

    E                   Request body validation error

Not sure what's wrong based on the error desc
Do you guys know a better way to see more exact error descriotion (what's wrong with the request data vs yaml?)

Openapi_core version 0.18.1

### Expected Behavior

Exected to isolate or relate what was not ok from request data vs yaml schema

### Steps to Reproduce

Not sure what is wrong with my request yet

### OpenAPI Core Version

0.18.1

### OpenAPI Core Integration

requests

### Affected Area(s)

_No response_

### References

_No response_

### Anything else we need to know?

_No response_

### Would you like to implement a fix?

None

--- a/openapi_core/validation/request/exceptions.py
+++ b/openapi_core/validation/request/exceptions.py
@@ -21,6 +21,8 @@ class RequestValidationError(ValidationError):
 
 class RequestBodyValidationError(RequestValidationError):
     def __str__(self) -> str:
+        if self.__cause__ is not None:
+            return f"Request body validation error: {self.__cause__}"
         return "Request body validation error"
 
 

--- a/tests/integration/validation/test_strict_json_validation.py
+++ b/tests/integration/validation/test_strict_json_validation.py
@@ -167,6 +167,31 @@ def test_request_validator_urlencoded_json_part_strict() -> None:
         validator.validate(request)
 
 
+def test_request_validator_error_message_includes_cause_details() -> None:
+    spec = _spec_schema_path()
+    validator = V30RequestValidator(spec)
+
+    request_json = {
+        "id": "123e4567-e89b-12d3-a456-426614174000",
+        "username": "Test User",
+        "age": "30",
+    }
+    request = MockRequest(
+        "http://example.com",
+        "post",
+        "/users",
+        content_type="application/json",
+        data=json.dumps(request_json).encode("utf-8"),
+    )
+
+    with pytest.raises(InvalidRequestBody) as exc_info:
+        validator.validate(request)
+
+    error_message = str(exc_info.value)
+    assert error_message.startswith("Request body validation error:")
+    assert "'30' is not of type 'integer'" in error_message
+
+
 def test_response_validator_strict_json_nested_types() -> None:
     """Test that nested JSON structures (arrays, objects) remain strict."""
     spec_dict = {

